pkg/hive/hive.go

```diff
@@ -66,7 +66,8 @@ type Service struct {
 	inLimiter         *ratelimit.Limiter
 	outLimiter        *ratelimit.Limiter
 	clearMtx          sync.Mutex
-	quit              chan struct{}
+	ctx               context.Context
```

Comment:
I recommend following the [documentation](https://github.com/golang/go/blob/master/src/context/context.go#L31-L37) and not abusing the context:
> Programs that use Contexts should follow these rules to keep interfaces
consistent across packages and enable static analysis tools to check context
propagation:
>
>**Do not store Contexts inside a struct type**; instead, pass a Context
explicitly to each function that needs it.